libvirt/resource_libvirt_volume.go

```diff
@@ -337,7 +337,31 @@ func resourceLibvirtVolumeRead(d *schema.ResourceData, meta interface{}) error {
 
 	volume, err := virConn.LookupStorageVolByKey(d.Id())
 	if err != nil {
-		return fmt.Errorf("Can't retrieve volume %s", d.Id())
+		virErr := err.(libvirt.VirError)
+		if virErr.Code == libvirt.VIR_ERR_NO_STORAGE_VOL {
+			volId := d.Id()
+			volPoolName := d.Get("pool").(string)
+			log.Printf("[INFO] Volume %s not found, attempting to start pool %s", volId, volPoolName)
+
+			volPool, err := virConn.LookupStoragePoolByName(volPoolName)
+			if err != nil {
+				return fmt.Errorf("Error retrieving pool %s for volume %s: %s", volPoolName, volId, err)
+			}
+			defer volPool.Free()
+
+			err = volPool.Create(0)
```

Comment:
> Also can you make sure the pool is not active (see this) before invoking Create?

It's not really needed (at least for directory-backed pools, which is the case I tested), still I added it.

> Can you replace 0 with libvirt.VIR_STORAGE_POOL_CREATE_NORMAL?

No, that constant does not exist in `libvirt-go`.